core/stores/sqlx/utils.go

```diff
@@ -81,10 +81,20 @@ func format(query string, args ...interface{}) (string, error) {
 					return "", err
 				}
 
-				// index starts from 1 for pg
-				if index > argIndex {
-					argIndex = index
+				switch ch {
```

Comment:
I think the original logic works for both '$' and ':'.

Reply:
the original logic test faild for "select name, age from users where bool=:1 and phone=:2 and phone2=:2"

Reply:
In this case, argIndex will be 3 with your code. But I think argIndex should be 2, right?

Reply:
argIndex will be 3 .  oracle test ok ,pg  test ok too ,but pg need more tests.